### ISSUE TITLE
XPSDrvSmpl insufficient deploy and test instructions

### ISSUE THREAD

**Issue by arsu-leo** [microsoft/Windows-driver-samples#461]
I've been following the instructions here:
https://github.com/microsoft/Windows-driver-samples/tree/master/print/XPSDrvSmpl

The first suspicious thing actually happens on build, where I get the following Warning:
>Severity	Code	Description	Project	File	Line	Suppression State
Warning		The driver package project C:\Users\MyUser\Documents\dev\myDriver\src\Windows-driver-samples\print\XPSDrvSmpl\Driver Package1\Driver Package1.vcxproj doesn't have any references to other projects and no projects were packaged.

Is that OK?

Even if able to compile, I don't know how to experiment and move forward.
I have configured a target deploy and have the Windbg running
"Debuggee is running..."

Now when I try to deploy the solution I get multiple errors that I'm unable to overcome.

The first one that I was able to handle is adding a "Test Certificate" for both "package" and "Driver Package" projects:
- For each of them, right click on properties on the solution explorer
- Go to Configuration properties, Driver Signing, General
- Set "Sign Mode" to "Test Sign"
- Set "Test Certificate" to "Create Test Certificate..."

Now, when I deploy, I get the following output:
> Deploying driver files for project "C:\Users\MyUser\Documents\dev\myDriver\src\Windows-driver-samples\print\XPSDrvSmpl\Driver Package1\Driver Package1.vcxproj".  Deployment may take a few minutes...
[15:46:23:459]: Remove Existing Remote Package
[15:46:23:519]: Task "Remove Existing Remote Package" completed successfully
[15:46:23:526]: Copy Driver Package
[15:46:24:065]: Copying file "C:\Users\MyUser\Documents\dev\myDriver\src\Windows-driver-samples\print\XPSDrvSmpl\x64\Debug\DriverPackage1.cer" to the target machine failed. The file could not be found.
[15:46:24:068]: ERROR: Task "Copy Driver Package" failed to complete successfully. Look at the logs in the driver test group explorer for more details on the failure.
[15:46:24:072]: Driver Removal
[15:46:24:072]: Removing any existing files from test execution folder.
[15:46:24:100]: Copying required files for "Driver Removal".
[15:46:25:300]: [Driver Removal] Command Line:
$KitRoot$\Testing\Runtimes\TAEF\te.exe "%SystemDrive%\DriverTest\Run\DriverTestTasks.dll" /select:"@Name='DriverTestTasks::_DriverRemoval'" /p:"InfFile=" /p:"ImportDriver=1" /p:"RemoveDriver=1" /p:"CertificateFile=DriverPackage1.cer" /p:"PackageGuid=x64" /p:"PrinterModel=xdsmpl" /rebootStateFile:%SystemDrive%\DriverTest\Run\DriverTestReboot.xml /enableWttLogging /wttDeviceString:$LogFile:file="%SystemDrive%\DriverTest\Run\Driver_Removal_00003.wtl",writemode=append,encoding=unicode,nofscache=true,EnableLvl="WexStartTest|WexEndTest|WexXml|WexProperty|WexCreateContext|WexCloseContext|*" /runas:Elevated
[15:46:28:507]: Result Summary: Total=1, Passed=0, Failed=1, Blocked=0, Warned=0, Skipped=0
[15:46:28:530]: ERROR: Task "Driver Removal" failed to complete successfully. Look at the logs in the driver test group explorer for more details on the failure.
[15:46:29:297]: Driver Preparation
[15:46:29:297]: Removing any existing files from test execution folder.
[15:46:29:325]: Copying required files for "Driver Preparation".
[15:46:30:222]: [Driver Preparation] Command Line:
$KitRoot$\Testing\Runtimes\TAEF\te.exe "%SystemDrive%\DriverTest\Run\DriverTestTasks.dll" /select:"@Name='DriverTestTasks::_DriverPreparation'" /p:"InfFile=" /p:"ImportDriver=1" /p:"RemoveDriver=1" /p:"CertificateFile=DriverPackage1.cer" /p:"PackageGuid=x64" /p:"PrinterModel=xdsmpl" /rebootStateFile:%SystemDrive%\DriverTest\Run\DriverTestReboot.xml /enableWttLogging /wttDeviceString:$LogFile:file="%SystemDrive%\DriverTest\Run\Driver_Preparation_00003.wtl",writemode=append,encoding=unicode,nofscache=true,EnableLvl="WexStartTest|WexEndTest|WexXml|WexProperty|WexCreateContext|WexCloseContext|*" /runas:Elevated
[15:46:33:504]: Result Summary: Total=1, Passed=0, Failed=1, Blocked=0, Warned=0, Skipped=0
[15:46:33:506]: ERROR: Task "Driver Preparation" failed to complete successfully. Look at the logs in the driver test group explorer for more details on the failure.
[15:46:34:147]: Default Printer Driver Package Installation Task
[15:46:34:147]: Removing any existing files from test execution folder.
[15:46:34:326]: Copying required files for "Default Printer Driver Package Installation Task".
[15:46:35:977]: [Default Printer Driver Package Installation Task] Command Line:
$KitRoot$\Testing\Runtimes\TAEF\te.exe "%SystemDrive%\DriverTest\Run\DefaultPrinterDriverPackageInstallationTask.dll" /select:"@Name='Microsoft.DriverKit.DefaultPrinterDriverPackageInstallationClass.PerformDefaultPrinterDriverPackageInstallation'" /p:"DQ=xdsmpl" /rebootStateFile:%SystemDrive%\DriverTest\Run\DriverTestReboot.xml /enableWttLogging /wttDeviceString:$LogFile:file="%SystemDrive%\DriverTest\Run\Default_Printer_Driver_Package_Installation_Task_00003.wtl",writemode=append,encoding=unicode,nofscache=true,EnableLvl="WexStartTest|WexEndTest|WexXml|WexProperty|WexCreateContext|WexCloseContext|*" /runas:Elevated
[15:46:37:151]: Result Summary: Total=1, Passed=0, Failed=1, Blocked=0, Warned=0, Skipped=0
[15:46:37:153]: ERROR: Task "Default Printer Driver Package Installation Task" failed to complete successfully. Look at the logs in the driver test group explorer for more details on the failure.
[15:46:37:734]: Driver Post Install Actions
[15:46:37:734]: Removing any existing files from test execution folder.
[15:46:37:788]: Copying required files for "Driver Post Install Actions".
[15:46:38:693]: [Driver Post Install Actions] Command Line:
$KitRoot$\Testing\Runtimes\TAEF\te.exe "%SystemDrive%\DriverTest\Run\DriverTestTasks.dll" /select:"@Name='DriverTestTasks::_DriverPostInstall'" /rebootStateFile:%SystemDrive%\DriverTest\Run\DriverTestReboot.xml /enableWttLogging /wttDeviceString:$LogFile:file="%SystemDrive%\DriverTest\Run\Driver_Post_Install_Actions_00003.wtl",writemode=append,encoding=unicode,nofscache=true,EnableLvl="WexStartTest|WexEndTest|WexXml|WexProperty|WexCreateContext|WexCloseContext|*" /runas:Elevated
[15:46:39:792]: Result Summary: Total=1, Passed=1, Failed=0, Blocked=0, Warned=0, Skipped=0
[15:46:39:794]: Task "Driver Post Install Actions" completed successfully
Driver Deployment Task Failed: Driver Removal
Driver Deployment Task Failed: Driver Preparation
Driver Deployment Task Failed: Default Printer Driver Package Installation Task

I  have tried different values and configuration ideas:
On project properties, for both "package" and "Driver Package", Driver install, deployment, optional arguments, i have used (no quotes): "xdsmpl", "%PathToInf", "XpsDrvSmpl", "package"

Solution properties, configuration properties configuration.
Swap the project  deploy checkboxes from "Driver Package" to "package", just to check.

Attacked an installation result log:
![driverInstallation](https://user-images.githubusercontent.com/7561829/75261472-631d0a00-57eb-11ea-9055-697cb540d28b.png)

If the deploy ever works, how would be able to test this? How can I move forward? I feel a high amount of missing info about a lot of topics, is there some extra resources I could look forward?

Greetings,

**Comment by wstaelens** [comment]
Yes the documentation could be improved. Hoping for the same... sometimes you have to spend hours to just find out Microsoft documentation is lacking info.

**Comment by v-junyli** [comment]
We are closing out this issue that unfortunately have not been addressed for a long time.  Our apologies for not attending to the issue in a timely manner.  If you have data that issue remains and it is still important, then feel free to reopen.
